app/src/main/java/com/github/sdpsharelook/TranslateActivity.kt

```diff
@@ -63,6 +67,10 @@ class TranslateActivity : AppCompatActivity() {
             val intent = Intent(ctx, NavigationMenuActivity::class.java)
             ctx.startActivity(intent)
         }
+        if(translateWord){
+            sourceText.text = intent.getStringExtra(TRANSLATOR_WORD)
+            translateWord = false
+        }

```

Comment:
wtf? there has to be a better way X(

Reply:
Yes, I was thinking to ask you how should we do this in a better way as we are going to do it for a few other activities.

Reply:
Probably should add a public function that can be called from other activities to set the translate text and translate it.
That way you can start the activity and call the function